### PR TITLE
Rewrite ObjectTracker, move it from a generated to a standalone layer

### PR DESCRIPTION
**Pull request by mark-lunarg**
Updated for device and instance data, for new coding styles; removed dead code; added WSI displayserver functions; and updated to match the style of other, newer layers.

Still a bit of clean-up going on related to the debugreport objects but I wanted to get some quick feedback.

### CODE REVIEW

**Comment by tobine** [comment]
Running this on our farm a couple of issues.

- On Linux Nvidia segF on CreateUnknownObject test
- Android build from Windows is hitting a clang++.exe error
- On Windows Nvidia CreateUnknownObject fails and appears to trigger all subsequent tests to fail

**Comment by mark-lunarg** [comment]
@tobine -- fixed the problems the LVTs.  Android builds on Windows crash for me on **Master**, linux works fine.  On Friday, Cody was also seeing master crash for android builds on Windows...

**Comment by tobine** [comment]
Still looking at android issues with this PR. Please refrain from pushing.

File: layers/object_tracker.h [hunk 1]
```diff
@@ -99,1078 +81,84 @@ struct instance_extension_enables {
...
-    }
-
-    lock.unlock();
-    if (!skipCall) {
-        get_dispatch_table(object_tracker_device_table_map, device)
-            ->FreeCommandBuffers(device, commandPool, commandBufferCount, pCommandBuffers);
-    }
-
-    lock.lock();
-    for (uint32_t i = 0; i < commandBufferCount; i++) {
```
Comment (stroyan) [inline layers/object_tracker.h]: These free_command_buffer calls need to be done before calling down to the next FreeCommandBuffers through the dispatch table.  The icd may immediately reissue a command buffer to a caller in another thread.  That will make
the free_command_buffer calls here too late.

Comment (mark-lunarg) [inline layers/object_tracker.h]: Same behavior was in FreeDescriptorSets, but all the other destroy/free APIs followed your example.  Changing both places.

**Comment by tobine** [comment]
Pushed branch tobin_gh738_obj_track_debug that adds a couple of commits to this branch, including one that fixes fails we were seeing on one arm32 platform.

**Comment by mark-lunarg** [comment]
Pushed to Master.

**Pull request closed without merging**